### PR TITLE
feat: Add invoice details

### LINKED ISSUE #408: Show invoice details when paying an invoice
Currently the user can only add the invoice as text, without any information about the contents of the invoice. At a very minimum the user should be able to 

- [ ] review the amount requested in the invoice 
- [ ] and counter party

### PR DESCRIPTION
**Pull request by holzeis**
invoice-details

resolves #408

### CODE REVIEW

**Review by bonomat — APPROVED** [review]
Very nice :)

File: lib/wallet/send.dart [hunk 1]
```diff
@@ -24,6 +29,83 @@ class _SendState extends State<Send> {
...
+          try {
+            invoice = await api.decodeInvoice(invoice: encodedInvoice);
+            setState(() {
+              invoice = invoice;
+            });
+          } catch (error) {
+            FLog.error(text: "Failed to decode invoice $encodedInvoice.", exception: error);
+            ScaffoldMessenger.of(context).showSnackBar(SnackBar(
+              backgroundColor: Colors.red,
+              content: Text("Failed to decode invoice. Error: " + error.toString()),
```
Comment (luckysori) [inline lib/wallet/send.dart:53]: 🔧 I think the user likely cannot make sense of the extra error context or do anything with that extra information. I would omit it and just say that they provided an invalid invoice.

**Review by luckysori — APPROVED** [review]
I really like how it's responsive to what the user pastes in!

File: lib/wallet/send.dart (hunk 1 above)
Comment (klochowicz) [reply lib/wallet/send.dart:53]: different perspective: IMHO at this stage of our product, any user feedback is helpful - if the user reports odd error message, then it's still good for us. And we also get more data doing internal testing, which is annoying, especially on mobile...

**Comment by klochowicz** [comment]
I'll go ahead with it, as I'm adding the QR code scanning feature on top of this.

**Pull request merged**